From 36482d61022f12b8f67fb5582efc4b6d772c17ca Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:15 -0400
Subject: [powerpc] powernv: Mark pnv_pci_init_ioda2_phb() as __init

Message-id: <1410545655-205645-27-git-send-email-dzickus@redhat.com>
Patchwork-id: 94068
O-Subject: [RHEL7 PATCH 026/626] powerpc/powernv: Mark pnv_pci_init_ioda2_phb() as __init
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 679750054ae8129a536734bccf526ff6da35376a
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Tue Jul 2 12:20:03 2013 -0600

    powerpc/powernv: Mark pnv_pci_init_ioda2_phb() as __init

    Mark pnv_pci_init_ioda2_phb() as __init.  It is called only from an
    init function (pnv_pci_init()), and it calls an init function
    (pnv_pci_init_ioda_phb()):

        pnv_pci_init                # init
          pnv_pci_init_ioda2_phb    # non-init
     pnv_pci_init_ioda_phb   # init

    This should fix a section mismatch warning.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/pci-ioda.c b/arch/powerpc/platforms/powernv/pci-ioda.c
index 7614aa1..2f6edd4 100644
--- a/arch/powerpc/platforms/powernv/pci-ioda.c
+++ b/arch/powerpc/platforms/powernv/pci-ioda.c
@@ -1285,7 +1285,7 @@ void __init pnv_pci_init_ioda_phb(struct device_node *np,
   pr_warning("  OPAL Error %ld performing IODA table reset !\n", rc);
 }
 
-void pnv_pci_init_ioda2_phb(struct device_node *np)
+void __init pnv_pci_init_ioda2_phb(struct device_node *np)
 {
  pnv_pci_init_ioda_phb(np, 0, PNV_PHB_IODA2);
 }
-- 
1.7.1